Specs/BSPropertySpec.mm

```diff
@@ -33,10 +41,24 @@
             });
         });
 
-        context(@"when the return type is a non-existent objective-c class", ^{
+        context(@"when the return type is an objective-c class conforming to a protocol", ^{
+            it(@"determines the protocol", ^{
+                property = [BSProperty propertyWithClass:[ClassWithProtocolInstance class] propertyNameString:@"protocolInstance"];
+                expect(property.returnType == [TestProtocolImpl class]).to(equal(YES));
+            });
+        });
+        
+        context(@"when the return type is an objective-c protocol", ^{
+            it(@"determines the protocol", ^{
+                property = [BSProperty propertyWithClass:[ClassWithProtocolProperty class] propertyNameString:@"protocolObject"];
+                expect(property.returnType == @protocol(TestProtocol)).to(equal(YES));
+            });
+        });
+        
+        context(@"when the return type conforms to multiple objective-c protocols", ^{
```

Comment:
I had a chat with Steven about this today. We concluded that the sensible thing would be to *not* throw an exception immediately in this situation, since it would prevent injection of such a property. It should instead require an explicit binding on that property, and throw an exception if none is provided.